visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -8,12 +8,15 @@ import { SettingsService } from "../../state/settingsService/settings.service"
 import { ColorConverter } from "../../util/color/colorConverter"
 import { CodeMapBuilding } from "./rendering/codeMapBuilding"
 
-export class CodeMapArrowService implements BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
+export class CodeMapArrowService
+	implements BuildingSelectedEventSubscriber, BuildingDeselectedEventSubscriber, BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
 	private VERTICES_PER_LINE = 5
 	private map: Map<String, Node>
 	private arrows: Object3D[]
 	private isHovered: boolean = false
+	private isClicked: boolean = false
```

Comment:
You don't need isClicked. If you want to check if something is clicked, you can simply check if `clickedNode` is not null or undefined.

```
if(clickedNode)
```
returns false, if clickedNode is undefined or null

Reply:
You can do that for isHovered as well.